cmd/publicize/server.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/git/v2"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+)
+
+type githubClient interface {
+	IsMember(org, user string) (bool, error)
+	CreateComment(owner, repo string, number int, comment string) error
+	GetPullRequest(org, repo string, number int) (*github.PullRequest, error)
+
+	ListIssueComments(owner, repo string, number int) ([]github.IssueComment, error)
+}
+
+var publicizeRe = regexp.MustCompile(`(?mi)^/publicize\s*$`)
+
+func helpProvider(_ []prowconfig.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The publicize plugin is used for merging a pull request into the bonded repository.`,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/publicize",
+		Description: "Merge the specific pull request into the bonded repository",
+		WhoCanUse:   "Members of the trusted organization for the repo.",
+		Examples:    []string{"/publicize"},
+	})
+	return pluginHelp, nil
+}
+
+type server struct {
+	githubTokenGenerator func() []byte
+
+	gitName  string
+	gitEmail string
+
+	bonds map[string]string
+
+	ghc githubClient
+	gc  git.ClientFactory
+
+	remoteHost string
+
+	dry bool
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) {
```

Comment:
The upstream expects to return an error. Why not this one?

Reply:
That's not true. see: https://github.com/kubernetes/test-infra/blob/master/prow/githubeventserver/githubeventserver.go#L110

Reply:
No, the upstream handlers all return errors. Please make the github event server  do it, too.

Reply:
I designed the githubevent server based on `hook`. for example: https://github.com/kubernetes/test-infra/blob/master/prow/hook/events.go#L120

But it will make sense for the handlers to return an error. I will follow-up on this.

Reply:
That's one layer away. `hook` the binary does not return an error. but each individual handler does. In this function here, you are implementing an individual handler and the server in one place, so I guess it makes sense